sunkit_image/granule.py

```diff
@@ -0,0 +1,391 @@
+            size = len(labeled_seg[labeled_seg == value])
+
+    # set all other 0 regions to mark value (1 or 0.5)
+    for value in values:
+        if np.sum(segmented_image[labeled_seg == value]) == 0:
+            if value != real_IG_value:
+                if not mark:
+                    segmented_image_fixed[labeled_seg == value] = 1
+                elif mark:
+                    segmented_image_fixed[labeled_seg == value] = 0.5
+
+    return segmented_image_fixed
+
+
+def mark_faculae(segmented_image, data, res):
+    """
+    Mark faculae seperatley from granules - give them a value of 2 not 1.
+    ----------
+    Parameters:
+        segmented_image (numpy array): the segmented image containing
+                                incorrect middles
+        data (numpy array): the original flux values
+        res (float): Spatial resolution (arcsec/pixel) of the data
+    ----------
+    Returns:
+        segmented_image_fixed (numpy array): the segmented image with faculae
+                                             marked as 1.5
+    """
+
+    fac_size_limit = 2  # max size of a faculae in sqaure arcsec
+    fac_pix_limit = fac_size_limit / res
+    fac_brightness_limit = np.mean(data) + 0.5 * np.std(data)
+
+    if len(np.unique(segmented_image)) > 3:
+        raise ValueError('segmented_image must have only values of 1, 0, ' +
+                         'an 0.5 (if dim centers marked)')
+
+    segmented_image = segmented_image
+    segmented_image_fixed = np.copy(segmented_image.astype(float))
+    labeled_seg = skimage.measure.label(segmented_image + 1, connectivity=2)
+    values = np.unique(labeled_seg)
+    for value in values:
+        mask = np.zeros_like(segmented_image)
+        mask[labeled_seg == value] = 1
+        # check that is a 1 (white) region
+        if np.sum(np.multiply(mask, segmented_image)) > 0:
+            region_size = len(segmented_image_fixed[mask == 1])
+            tot_flux = np.sum(data[mask == 1])
+            # check that region is small
+            if region_size < fac_pix_limit:
+                # check that avg flux very high
+                if tot_flux / region_size > fac_brightness_limit:
+                    segmented_image_fixed[mask == 1] = 1.5
+
+    return segmented_image_fixed
+
+def kmeans_segment(data, llambda_axis=-1):
+    """kmeans clustering: uses a kmeans algorithm to cluster data,
+       in order to independently verify the skimage clustering method
+       (e.g using cross_correlation() below).
+        ----------
+       Parameters:
+            data (numpy array): data to be clustered
+            llambda_axis (int): index for wavelength, -1 if scalar array.
+        ----------
+        Returns:
+            labels (numpy array): an array of labels, with 0 = granules,
+                                  2 = intergranules, 1 = in-between.
+            """
+
+    if llambda_axis not in [-1, 2]:
+        raise Exception('Wrong data shape. \

```

Comment:
This should be a more focused Exception, I think a ValueError would be more correct but I need to check the list of exceptions to be certain. 

Also if the value can not be -1 or 2, should this be able to be changed by a user?

Reply:
I am not sure why this is here (this is one function I did not write); I believe it would indeed make more sense to just hardcode that to -1. I will try to ascertain why it was written that way.